packages/lit-analyzer/src/test/rules/no-unclosed-tag.ts

```diff
@@ -7,7 +7,49 @@ tsTest("Report unclosed tags", t => {
 	hasDiagnostic(t, diagnostics, "no-unclosed-tag");
 });
 
```

Comment:
In an ideal world, we might want to allow:

```html
<p>One
<p>Two
```

Or just:

```html
<p>Hi
```

And we'd want to put warnings on dangling closing tags that don't match up to any opening tags.

All of those may be somewhat tricky to do with how parse5 represents the tree though, requiring us to notice spans in the original text that aren't represented in the parsed tree.

Reply:
These are very tricky, and another good case to add to an issue for followup.

Philosophically though, I wonder if the lit-analyzer should avoid HTML diagnostics and leave that to an HTML specific tool. It may even be worth scoping the `no-unclosed-tag` to only check custom element tags.